client/network/src/discovery.rs

```diff
@@ -730,24 +687,28 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 							}
 						},
 					},
-					KademliaEvent::OutboundQueryCompleted {
+					KademliaEvent::OutboundQueryProgressed {
 						result: QueryResult::GetRecord(res),
 						stats,
+						step,
 						..
 					} => {
 						let ev = match res {
-							Ok(ok) => {
-								let results = ok
-									.records
-									.into_iter()
-									.map(|r| (r.record.key, r.record.value))
-									.collect();
-
-								DiscoveryOut::ValueFound(
-									results,
-									stats.duration().unwrap_or_default(),
-								)
-							},
+							Ok(ok) =>
+								if let GetRecordOk::FoundRecord(r) = ok {
+									DiscoveryOut::ValueFound(
+										vec![(r.record.key, r.record.value)],
```

Comment:
not sure this is correct. Are we expecting multiple records here or just one?

NOTE: OutboundQueryCompleted was renamed OutboundQueryProgressed and now emits multiple events (last one indicated by `step.last: true`)

Reply:
https://github.com/paritytech/substrate/pull/12734/commits/dbb7becf17e81a39519f2a8d67f36d49845d5f51